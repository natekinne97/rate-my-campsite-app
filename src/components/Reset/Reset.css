.reset-password{
    text-align: center;
}