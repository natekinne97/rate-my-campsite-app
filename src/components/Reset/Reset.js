import React from 'react';
import authApi from '../../api-services/auth-api';

class Reset extends React.Component{

    // get the params
    static defaultProps = {
        match: { params: {} },
    }

    state={
        user_name: null,
        passwordError: null,
        error: null
    }

    // redirects user to the login page when succesful
    redirectLogin(){
        const { history } = this.props
        history.push('/login');
    }


    // get the username first display it on the screen and use it to change the pass
    componentDidMount(){
        console.log('calling rerender');
        const { token } = this.props.match.params
        authApi.getUsername(token)
            .then(user=>{
                console.log(user, 'user');
                if (user === 'password reset link is invalid or has expired'){
                    this.setState({
                        error: user
                    })
                }else{
                    this.setState({
                        user_name: user
                    })
                }
                
            })
    }

    // check passwords match
    repeatOnChange = () => {
        const og = document.getElementById('password_reset').value;

        const repeat_password = document.getElementById('pass_repeat').value;

        if (og !== repeat_password) {
            this.setState({
                passwordError: 'passwords do not match'
            })
        } else {
            this.setState({
                passwordError: null
            })
        }
    }

    handleSubmit = e =>{
        e.preventDefault();
        const {password, repeat_pass} = e.target;
        console.log(this.state.user_name, 'user name is');
        authApi.updatePassword( this.state.user_name ,repeat_pass.value)
            .then(res=>{
                if (res === 'update succesful'){
                    password.value = '';
                    repeat_pass.value = '';
                    // redirect to login
                    this.redirectLogin();

                }else{
                    this.setState({
                        passwordError: res.error
                    })
                }
            })

    }

    // displays only the form
    renderResetPasswordPage(){
        return(
           
                <form className="gen-form" onSubmit={this.handleSubmit}>
                    {this.state.passwordError
                    ? <p className="red">{this.state.passwordError}</p>
                    : null}
                    <label>New password</label>
                    <input id="password_reset" type="password" name="password" required/>

                    <label>Repeat password</label>
                    <input id="pass_repeat" type="password" name="repeat_pass" onChange={this.repeatOnChange}/>

                    <button type="submit">Submit</button>
                </form>
        )
    }


    render(){
        return(
            <div className="resest-password">
                <header><h1>{this.state.user_name}</h1></header>
                {this.state.error
                ? <p>{this.state.error}</p>
                : this.renderResetPasswordPage()}
            
            </div>
        )
    }

}

export default Reset;