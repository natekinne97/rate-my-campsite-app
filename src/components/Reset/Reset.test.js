import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Reset from './Reset';

describe.only('Review form component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Reset />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Reset />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the UI as expected with no unreads', () => {
        const tree = renderer
            .create(<Reset />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});