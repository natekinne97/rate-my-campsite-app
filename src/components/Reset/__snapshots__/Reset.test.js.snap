// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review form component renders the UI as expected 1`] = `
<div
  className="resest-password"
>
  <header>
    <h1 />
  </header>
  <form
    className="gen-form"
    onSubmit={[Function]}
  >
    <label>
      New password
    </label>
    <input
      id="password_reset"
      name="password"
      required={true}
      type="password"
    />
    <label>
      Repeat password
    </label>
    <input
      id="pass_repeat"
      name="repeat_pass"
      onChange={[Function]}
      type="password"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`Review form component renders the UI as expected with no unreads 1`] = `
<div
  className="resest-password"
>
  <header>
    <h1 />
  </header>
  <form
    className="gen-form"
    onSubmit={[Function]}
  >
    <label>
      New password
    </label>
    <input
      id="password_reset"
      name="password"
      required={true}
      type="password"
    />
    <label>
      Repeat password
    </label>
    <input
      id="pass_repeat"
      name="repeat_pass"
      onChange={[Function]}
      type="password"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
